src/bomsquad/vulndb/db/ingest.py

```diff
@@ -19,21 +19,34 @@ class _NVDResultGen:
     result_sets: Generator[NVDResultSet[Any], None, None]
     current: NVDResultSet[Any]
     first_ts: datetime
+    empty: bool
 
     def __init__(self, result_sets: Generator[NVDResultSet[Any], None, None]) -> None:
         self.result_sets = result_sets
-        self.current = next(self.result_sets)
-        self.first_ts = self.current.timestamp
+        try:
+            self.current = next(self.result_sets)
+            self.first_ts = self.current.timestamp
+            self.empty = False
+        except StopIteration:
+            self.empty = True
+            pass
 
     def __iter__(self) -> Iterator[Any]:
         return self
 
     def __next__(self) -> Any:
+        if self.empty:
+            raise StopIteration
         try:
             return next(self.current)
         except StopIteration:
-            self.current = next(self.result_sets)
-            return next(self)
```

Comment:
There's a problem here if the next result set is empty. The simplest answer is probably to recurse __next__ after advancing the result_sets iterator.